ir/src/integrity_stmts/constraint.rs

```diff
@@ -0,0 +1,77 @@
+use super::NodeIndex;
+
+/// A [ConstraintRoot] represents the entry node of a subgraph representing an integrity constraint
+/// within the [AlgebraicGraph]. It also contains the row offset for the constraint which is the
+/// maximum of all row offsets accessed by the constraint. For example, if a constraint only
+/// accesses the trace in the current row then the row offset will be 0, but if it accesses the
+/// trace in both the current and the next rows then the row offset will be 1.
```

Comment:
I think this description may need to be updated.

Reply:
Good catch, thanks! I'll update it when I move the boundary constraints in, as it will need to be updated at that time anyway.